task_sdk/tests/execution_time/test_task_runner.py

```diff
@@ -19,20 +19,44 @@
 
 import uuid
 from datetime import timedelta
-from pathlib import Path
 from socket import socketpair
 from unittest import mock
 
 import pytest
+from kgb import SpyAgency
 from uuid6 import uuid7
 
+from airflow.exceptions import AirflowSkipException
 from airflow.sdk import DAG, BaseOperator
 from airflow.sdk.api.datamodels._generated import TaskInstance, TerminalTIState
 from airflow.sdk.execution_time.comms import DeferTask, StartupDetails, TaskState
-from airflow.sdk.execution_time.task_runner import CommsDecoder, parse, run
+from airflow.sdk.execution_time.task_runner import CommsDecoder, RuntimeTaskInstance, parse, run
 from airflow.utils import timezone
 
 
+def mocked_parse(what: StartupDetails, dag, task_id) -> RuntimeTaskInstance:
+    """A mock parse function to support in line DAGs."""
+    # Retrieve the task from the DAG
+    task = dag.task_dict[task_id]
+    return RuntimeTaskInstance.model_construct(**what.ti.model_dump(exclude_unset=True), task=task)
+
+
+class CustomOperator(BaseOperator):
+    def execute(self, context):
+        task_id = context["task_instance"].task_id
+        print(f"Hello World {task_id}!")
+
+
+def get_inline_dag(dag_id: str, tasks: BaseOperator) -> DAG:
+    dag = DAG(
+        dag_id=dag_id,
+        default_args={"start_date": timezone.datetime(2024, 12, 3)},
+    )
+    setattr(tasks, "dag", dag)
```

Comment:
Nit: `task` since it's one, not a list. (The `setattr(tasks` line was confusing me)

```suggestion
def get_inline_dag(dag_id: str, task: BaseOperator) -> DAG:
    dag = DAG(
        dag_id=dag_id,
        default_args={"start_date": timezone.datetime(2024, 12, 3)},
    )
   task.dag = dag
```

Reply:
Yeah valid point.